pepper-apis/studybuilder-cli/src/main/java/org/broadinstitute/ddp/studybuilder/task/osteo/OsteoV2ReleasePdfUpdates.java

```diff
@@ -0,0 +1,104 @@
+import org.broadinstitute.ddp.model.pdf.PdfConfigInfo;
+import org.broadinstitute.ddp.model.pdf.PdfVersion;
+import org.broadinstitute.ddp.studybuilder.PdfBuilder;
+import org.broadinstitute.ddp.studybuilder.task.CustomTask;
+import org.jdbi.v3.core.Handle;
+import org.jdbi.v3.sqlobject.SqlObject;
+import org.jdbi.v3.sqlobject.customizer.Bind;
+import org.jdbi.v3.sqlobject.statement.SqlUpdate;
+
+import java.io.File;
+import java.nio.file.Path;
+import java.util.List;
+
+@Slf4j
+public class OsteoV2ReleasePdfUpdates implements CustomTask {
+
+    private static final String DATA_FILE = "patches/osteo-v2-release-pdfs.conf";
+    private static final String STUDY_GUID = "CMI-OSTEO";
+
+    private Path cfgPath;
+    private Config studyCfg;
+    private Config dataCfg;
+    private OsteoV2ReleasePdfUpdates.SqlHelper sqlHelper;
+
+    @Override
+    public void init(Path cfgPath, Config studyCfg, Config varsCfg) {
+        if (!studyCfg.getString("study.guid").equals(STUDY_GUID)) {
+            throw new DDPException("This task is only for the " + STUDY_GUID + " study!");
+        }
+        this.cfgPath = cfgPath;
+        this.studyCfg = studyCfg;
+
+        File file = cfgPath.getParent().resolve(DATA_FILE).toFile();
+        if (!file.exists()) {
+            throw new DDPException("Data file is missing: " + file);
+        }
+        this.dataCfg = ConfigFactory.parseFile(file).resolveWith(varsCfg);
+    }
+
+    @Override
+    public void run(Handle handle) {
+        this.sqlHelper = handle.attach(OsteoV2ReleasePdfUpdates.SqlHelper.class);
+        StudyDto studyDto = handle.attach(JdbiUmbrellaStudy.class).findByStudyGuid(studyCfg.getString("study.guid"));
+        UserDto adminUser = handle.attach(JdbiUser.class).findByUserGuid(studyCfg.getString("adminUser.guid"));
+
+        List<? extends Config> pdfs = dataCfg.getConfigList("pdfs");
+        pdfs.forEach(pdf -> cleanupV2Version(handle, pdf.getString("name"), studyDto.getId(), "v2"));
+
+        //re-insert new v2 version
+        var pdfBuilder = new PdfBuilder(cfgPath.getParent(), studyCfg, studyDto, adminUser.getUserId());
+        pdfs.forEach(pdf -> pdfBuilder.insertPdfConfig(handle, pdf));
+
+    }
+
+    public void cleanupV2Version(Handle handle, String pdfConfigName, long studyId, String versionTag) {
+
+        PdfDao pdfDao = handle.attach(PdfDao.class);
+        PdfConfigInfo info = pdfDao.findConfigInfoByStudyIdAndName(studyId, pdfConfigName)
+                .orElseThrow(() -> new DDPException("Could not find pdf with name=" + pdfConfigName));
+
+        PdfVersion version = pdfDao.findOrderedConfigVersionsByConfigId(info.getId())
+                .stream()
+                .filter(ver -> ver.getVersionTag().equals(versionTag))
+                .findFirst()
+                .orElseThrow(() -> new DDPException("Could not find pdf version with data source for versionTag=" + versionTag));
+
+        //delete this version to do clean insert
```

Comment:
At some point we should invest time to revive StudyBuilder code/functionality to just insert new pdf templates (raw bytes) . It helps if no pdf config changes are needed . Here I have some pdf config changes hence doing clean up and re-inserting pdf config fr v2 along with new updated pdf templates